alot/db/thread.py

```diff
@@ -151,13 +151,14 @@ def get_authors(self):
         :rtype: list of (str, str)
         """
         if self._authors is None:
+            # Sort messages with date first (by date descending), and those
+            # without a date last.
+            msgs = sorted(
```

Comment:
very elegant! I'll have to test this but having only read the code, I like the change.